lib/uart.cpp

```diff
@@ -76,7 +85,14 @@ uint8_t Uart::read(Uart::Address register_address) {
             if (this->dlab()) {//DLL
                 return this->m_dll;
             } else {//RHR
-                return std::getchar();//TODO is the fact that this is blocking a problem?
+                uint8_t data;
+                if(receive_queue.size()>0){
+                    data = receive_queue.front();
+                    receive_queue.pop();
+                }else{
+                    assert(false && "Tried to read from empty input");
```

Comment:
I'm cool with this assert failing if software reads from the register since it will be good at helping us catch software / kernel bugs.

A comment clarifying this wouldn't hurt. Or perhaps change the message to "Buggy software tried to read from empty input" to accomplish a similar effect. :)

Reply:
Yeah, would def be good to clarify since from my understanding, we've been using assertions to find bugs in the emulator, not in software running on it

Reply:
Agreed, and then once the kernel is working and RVSW stuff is more stabilized, we can remove it and probably just return 0.

Reply:
Good point. I've added a comment in my most recent commit.